starlette/endpoints.py

```diff
@@ -31,3 +32,40 @@ def __init__(self, scope: Scope):
         if "app" in self.scope:
             raise HTTPException(status_code=405)
         return PlainTextResponse("Method Not Allowed", status_code=405)
+
+
+class WebSocketEndpoint:
+    def __init__(self, scope: Scope):
+        self.scope = scope
+        self.websocket = None
+        self.close_code = None
+        self.kwargs = None
```

Comment:
I am unresolving this for the moment - just curious about the reason why we should not store the websocket in the instance.

I am finding the initial pattern that allowed referencing `self.websocket` to be a lot more flexible towards implementing specific subclass behavior, and it would also allow us to have something like `self.accept()` which could conveniently wrap the `websocket.accept()` method used in the `on_connect` method without having to explicitly reference the websocket instance.

Thoughts?

Reply:
Will resolve this and continue discussing in the issue later, a bit obscured here.